apps/emqx_rule_engine/README.md

```diff
@@ -1,23 +1,46 @@
 
-# emqx-rule-engine
+# Emqx Rule Engine
 
-IoT Rule Engine
+The rule engine's goal is to provide a simple and flexible way to transform and
+reroute the messages coming to the EMQX broker. For example, one message
+containing measurements from multiple sensors of different types can be
+transformed into multiple messages.
+
+
+## Concepts
+
+A rule is quite simple. A rule describes which messages it affects by
+specifying a topic filter and a set of conditions that need to be met. If a
+message matches the topic filter and all the conditions are met, the rule is
+triggered. The rule can then transform the message and route it to a different
+topic, or send it to another service (defined by an EMQX bridge). The rule
+engine's message data transformation is designed to work well with JSON
+formatted data.
```

Comment:
```suggestion
engine's message data transformation is designed to work well with structured data
such as JSON, avro, and protobuf.
```

Reply:
schema registry is not added yet, but rule-engine should be able to work with any kind of structured data as long as they can be deserialized into an Erlang map. (maybe add this info to the readme).